app/controllers/api/subcollections/physical_racks.rb

```diff
@@ -0,0 +1,9 @@
+module Api
+  module Subcollections
+    module PhysicalRacks
+      def physical_racks_query_resource(object)
+        object.respond_to?(:physical_rack) ? PhysicalRack.where(:id => object.physical_rack_id) : []
```

Comment:
can't this be object.physical_rack  instead of the PhysicalRack.where ... ?

Reply:
Actually no, because PhysicalRack has many PhysicalStorages and not the inverse. So if we call `object.physical_rack` the code will break. That's why, instead, we use the reference `physical_rack_id` which denotes the PhysicalRack that the PhysicalStorage belongs to.

Reply:
@EsdrasVP What will break?  It looks like the relationships are defined in both directions...
```
app/models/physical_chassis.rb:9:  belongs_to :physical_rack, :foreign_key => :physical_rack_id, :inverse_of => :physical_chassis
app/models/physical_rack.rb:4:  belongs_to :ext_management_system, :foreign_key => :ems_id, :inverse_of => :physical_racks,
app/models/physical_rack.rb:6:  has_many :physical_chassis, :dependent => :nullify, :inverse_of => :physical_rack
app/models/physical_rack.rb:7:  has_many :physical_servers, :dependent => :nullify, :inverse_of => :physical_rack
app/models/physical_rack.rb:8:  has_many :physical_storages, :dependent => :nullify, :inverse_of => :physical_rack
app/models/physical_server.rb:22:  belongs_to :physical_rack, :foreign_key => :physical_rack_id, :inverse_of => :physical_servers
app/models/physical_storage.rb:4:  belongs_to :physical_rack, :foreign_key => :physical_rack_id, :inverse_of => :physical_storages
```

Reply:
@bdunne They have, PhysicalRack and PhysicalChassis will have many PhysicalStorages, but the problem is that PhysicalStorage will only belong to one PhysicalRack/PhysicalChassis. So what happens when I use `object.physical_rack` is that a method `all` is called to return a list of PhysicalRacks/PhysicalChassis when this list doesn't exist and only a object is returned. So when I access `/api/physical_storages/:id/physical_racks`, for instance, the following error is presented:

```
{
    "error": {
        "kind": "internal_server_error",
        "message": "undefined method `all' for #<PhysicalRack:0x007f1bf0c52098>",
        "klass": "NoMethodError"
    }
}
```

Reply:
if there's one and only one, the subcollection does not make sense here, it can currently be accessed via GET /api/physical_storages/:id?attributes=physical_rack
as shown by the OPTIONS /api/physical_storages in the relationships section.

Reply:
> if there's one and only one, the subcollection does not make sense here, it can currently be accessed via GET /api/physical_storages/:id?attributes=physical_rack
> as shown by the OPTIONS /api/physical_storages in the relationships section.

@abellotti @bdunne This makes sense, I spoke with @rodneyhbrown7 and we agreed to use the current method to access the physical storage relationships. So this PR can be closed.